### ISSUE TITLE
Finding authentication details for official providers is difficult

### ISSUE THREAD

**Issue by jbw976** [upbound/docs#209]
I thought there was an existing issue for this, but I can't find it, so I'm capturing it here to stay organized 😇 

Users of the official providers often ask about how to configure authentication for the provider.  These resources seem to be a bit scattered, e.g.:

* provider-aws
  * https://github.com/upbound/provider-aws/blob/main/AUTHENTICATION.md: really good material, but no one can ever find this one their own
  * https://marketplace.upbound.io/providers/upbound/provider-aws/latest/docs/configuration, a subset of the AUTHENTICATION.md file and hard to parse since it's embedded in the provider marketplace docs as opposed to a rich dedicated docs site
* provider-azure
  * https://github.com/upbound/provider-azure/blob/main/AUTHENTICATION.md: same discoverability problem as provider-aws AUTHENTICATION.md file, no can find this
  * https://marketplace.upbound.io/providers/upbound/provider-azure/latest/docs/configuration: the provider docs don't seem to mention any other options for configuring Azure auth
* provider-gcp
  * is missing a rich AUTHENTICATION.md file, so there doesn't appear to be an option for exploring all the auth mechanisms at all
  * https://marketplace.upbound.io/providers/upbound/provider-gcp/latest/docs/configuration: has some good content, but could benefit from a rich dedicated docs experience

Users would benefit if there was in general:

1. an easy place to find the authentication methods for each provider
2. that place had a rich docs experience with examples, "on this page" navigation table, etc.

**Comment by plumbis** [comment]
AWS resolved with #206

**Comment by tr0njavolta** [comment]
All clouds accounted for! @jbw976 I'll close if this looks good.

**Comment by jbw976** [comment]
amazing, the https://docs.upbound.io/ are looking really good now!! please do take the honors to close this out @tr0njavolta 🙇‍♂️
